datafusion-cli/src/object_storage.rs

```diff
@@ -36,10 +42,29 @@ pub fn get_s3_object_store_builder(
         builder = builder
             .with_access_key_id(access_key_id)
             .with_secret_access_key(secret_access_key);
-    }
 
-    if let Some(session_token) = cmd.options.get("session_token") {
-        builder = builder.with_token(session_token);
+        if let Some(session_token) = cmd.options.get("session_token") {
+            builder = builder.with_token(session_token);
+        }
+    } else {
+        let config = aws_config::from_env().load().await;
```

Comment:
Unlike before, this now ensures that users get the full coherent configuration experience that they might expect from a CLI tool.

Reply:
cc @r4ntix  as I think you contributed some of the previous configuration code

Reply:
> cc @r4ntix as I think you contributed some of the previous configuration code

@alamb Thanks for the heads up, I think this is a great improvement. Thanks @tustvold  👏